src/overlays/actors/ovl_En_Md/z_en_md.c

```diff
@@ -21,79 +57,813 @@ const ActorInit En_Md_InitVars = {
+        this->unk_212 = 1;
+        this->unk_1E0.unk_00 = 0;
+        this->actionFunc = func_80AABD0C;
+        this->actor.speedXZ = 1.5f;
+        return;
+    }
+
+    if (this->skelAnime.animCurrentSeg == &D_060002C8) {
+        func_80034F54(globalCtx, &this->unk_214, &this->unk_236, 17);
+    }
+
+    if ((this->unk_1E0.unk_00 == 0) && (globalCtx->sceneNum == SCENE_SPOT10)) {
+        if ((s32)(player->stateFlags2 << 7) < 0) {
+            player->stateFlags2 |= 0x2000000;
+            player->unk_6A8 = &this->actor;
+            func_8010BD58(globalCtx, 0x22);
+            this->actionFunc = func_80AABC10;
+            return;
+        }
+
+        if (this->actor.xzDistFromLink < (30.0f + this->collider.dim.radius)) {
+            player->stateFlags2 |= 0x800000;
+        }
+    }
+}
+
+void func_80AABC10(EnMd* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (globalCtx->msgCtx.unk_E3EE >= 4) {
+        this->actionFunc = func_80AAB948;
+        globalCtx->msgCtx.unk_E3EE = 4;
+    } else if (globalCtx->msgCtx.unk_E3EE == 3) {
+        Audio_PlaySoundGeneral(NA_SE_SY_CORRECT_CHIME, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        this->actor.textId = 0x1067;
+        func_8002F2CC(&this->actor, globalCtx, this->collider.dim.radius + 30.0f);
+
+        this->actionFunc = func_80AAB948;
+        globalCtx->msgCtx.unk_E3EE = 4;
+    } else {
+        player->stateFlags2 |= 0x800000;
+    }
+}
+
+void func_80AABD0C(EnMd* this, GlobalContext* globalCtx) {
+    func_80034F54(globalCtx, &this->unk_214, &this->unk_236, 17);
+    func_80AAA93C(this);
+
+    if (!(func_80AAB370(this, globalCtx)) || (this->unk_212 != 0)) {
+        this->actor.shape.rot = this->actor.posRot.rot;
+        return;
+    }
+
+    if (CHECK_QUEST_ITEM(QUEST_KOKIRI_EMERALD) && !(gSaveContext.eventChkInf[1] & 0x1000) && (globalCtx->sceneNum == SCENE_SPOT04)) {
+        func_80106CCC(globalCtx);
+        gSaveContext.eventChkInf[1] |= 0x1000;
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    func_80AAA92C(this, 11);
+
+    this->skelAnime.animPlaybackSpeed = 0.0f;
+    this->actor.speedXZ = 0.0f;
+    this->actor.initPosRot.pos = this->actor.posRot.pos;
+    this->actionFunc = func_80AAB8F8;
+}
+
+void EnMd_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnMd* this = THIS;
+    s32 pad;
+
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    EnMd_UpdateEyes(&this->actor);
+    func_80AAB5A4(&this->actor, globalCtx);
+    Actor_MoveForward(&this->actor);
+    func_80AAB158(&this->actor, globalCtx);
+    func_8002E4B4(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    this->actionFunc(&this->actor, globalCtx);
+}
+
+s32 func_80AABEF0(struct GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                  struct Actor* thisx, Gfx** gfx) {
```

Comment:
This function is normally named <Actor>_OverrideLimbDraw, and the `struct` keyword should be omitted for its arguments.
```suggestion
s32 EnMd_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
                          Actor* thisx, Gfx** gfx) {
```